open-sphere-plugins/feature-actions/src/main/java/io/opensphere/featureactions/controller/FeatureActionStateController.java

```diff
@@ -0,0 +1,260 @@
+            newFeatureAction.setName(featureActionType.getTitle() + " - " + id);
+            newFeatureAction.setEnabled(featureActionType.isActive());
+            newFeatureAction.setGroupName(featureActionType.getDescription());
+            WFS100FilterToDataFilterConverter filterConverter = new WFS100FilterToDataFilterConverter();
+            newFeatureAction.setFilter(filterConverter.apply(featureActionType.getFilter()));
+            StyleAction style = new StyleAction();
+            FeatureStyleActionType act = (FeatureStyleActionType)featureActionType.getActions().getAction().get(0).getValue();
+            style.getStyleOptions().setColor(ColorUtilities.convertFromHexString(act.getColor(), 1, 2, 3, 0));
+            style.getStyleOptions().setSize(act.getSize().intValue());
+            newFeatureAction.getActions().add(style);
+            myRegistry.add(featureActionType.getType(), Collections.singleton(newFeatureAction), this);
+            addResource(id, new Pair<>(newFeatureAction, featureActionType.getType()));
+        }
+    }
+
+    @Override
+    public boolean canActivateState(Node node)
+    {
+        return StateXML.anyMatch(node, "/:state/:featureActions/:featureAction");
+    }
+
+    @Override
+    public boolean canActivateState(StateType state)
+    {
+        return !StateUtilities.getFeatureActions(state).getFeatureAction().isEmpty();
+    }
+
+    @Override
+    public boolean canSaveState()
+    {
+        Set<DataTypeInfo> dataTypeSet = getFeatureLayers();
+        for (DataTypeInfo dataType : dataTypeSet)
+        {
+            if (!myRegistry.get(dataType.getTypeKey()).isEmpty())
+            {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    @Override
+    protected void deactivate(Pair<FeatureAction, String> resource)
+    {
+        myRegistry.remove(resource.getSecondObject(), Collections.singleton(resource.getFirstObject()), this);
+    }
+
+    @Override
+    public boolean isSaveStateByDefault()
+    {
+        return true;
+    }
+
+    @Override
+    public void saveState(Node node)
+    {
+        Document doc = node instanceof Document ? (Document)node : node.getOwnerDocument();
+        Set<DataTypeInfo> dataTypeSet = getFeatureLayers();
+        for (DataTypeInfo dataType : dataTypeSet)
+        {
+            List<FeatureAction> actionList = myRegistry.get(dataType.getTypeKey());
+            for (FeatureAction currentAction : actionList)
+            {
+                try
+                {
+                    Node baseNode = StateXML.createChildNode(node, doc, node, "/" + ModuleStateController.STATE_QNAME + "/:featureActions",
+                            "featureActions");
+                    Node featureActionNode = StateXML.createChildNode(baseNode, doc, baseNode,
+                            "/" + ModuleStateController.STATE_QNAME + "/:featureActions/:featureAction", "featureAction");
```

Comment:
I would recommend making a FeatureActionStateConstants class with these string values (similar to CSVStateConstants, StateConstants, etc.)